builders/xcm/xcm-sdk.md

```diff
@@ -0,0 +1,713 @@
+```
+
+### Deposit {: #deposit }
+
+To deposit an asset to Moonbeam from another network, you'll have to first build the request using information from the origin chain before you can send the request. To do so, you'll take the following steps:
+
+1. Call the `deposit` function and pass in the asset symbol for the asset to be deposited. This will return a `chains` array containing the asset's origin network information and a `from` function which will be used to build the request
+2. Call the `from` function and pass in the chain key of the origin network. Then, call `get` and pass in the address of the account on Moonbeam you want to deposit the funds to and pass in the [Polkadot signer](#creating-signers). This will return a `send` function which already contains all the necessary info to perform the deposit, and it is used in the next step. In addition, other elements such as information about the origin chain asset, the `xc` representation of the asset on Moonbeam, are returned and might be important for logging purposes
+3. The `send` function is used to send the built deposit request along with the amount to send. You can optionally provide a callback function to handle the extrinsic events
+
+To obtain some of the data required to build the deposit request, such as the asset symbol and chain key of the origin network, you can import `AssetSymbol` and `ChainKey` from the `@moonbeam-network/xcm-config` package.
+
+An example of the steps described above to deposit DOT from the Polkadot relay chain to xcDOT on Moonbeam is as follows:
+
+```js
+import { AssetSymbol, ChainKey } from '@moonbeam-network/xcm-config';
+
+async function deposit() {
+  const dot = AssetSymbol.DOT;
+  const polkadot = ChainKey.Polkadot;
+
+  const { chains, from } = moonbeam.deposit(dot);
+
+  console.log(
+    `\nYou can deposit ${dot} from these chains: `,
+    chains.map((chain) => chain.name),
+  );
+
+  const { asset, sourceBalance, source, min, send } = await from(polkadot).get(
+    'INSERT-MOONBEAM-ADDRESS',
+    polkadotSigner,
+  );
+
+  console.log(
+    `Your ${asset.originSymbol} balance in ${source.name}: ${toDecimal(
+      sourceBalance,
+      asset.decimals,
+    )}. Minimum transferable amount is: ${toDecimal(min, asset.decimals)}`,
+  );
+
+  await send('INSERT-AMOUNT', (event) => console.log(event));
+}
+
+deposit();
+```
+
+As previously mentioned, the `deposit` function returns a `chains` array and a `from` function. The `chains` array corresponds to the chains you can deposit the given asset from (for the asset that was initially passed into the `deposit` function). An example of the `chains` array is as follows:
+
+```js
+chains: [
+  {
+    key: 'Polkadot',
+    name: 'Polkadot',
+    ws: 'wss://rpc.polkadot.io',
+    weight: 1000000000,
+    parachainId: 0
+  }
+]
+```
+
+#### From {: #from }
+
+The `from` function requires a chain key to be passed into it for the origin chain in which the assets are sent from and returns a `get` function. 
```

Comment:
@eshaben here is what we discussed with you about `AssetSymbol` vs `Asset` and `ChainKey` vs `Chain`. I already changed this in the SDK so now you can pass both.
```typescript
  deposit: (
    symbolOrAsset: Symbols | Asset<Symbols>,
  ) => XcmSdkDeposit<Symbols, ChainKeys>;
  withdraw: (
    symbolOrAsset: Symbols | Asset<Symbols>,
  ) => XcmSdkWithdraw<Symbols, ChainKeys>;
```
```typescript
  from: (
    keyOrChain: ChainKeys | Chain<ChainKeys>,
  ) => XcmSdkDepositFrom<Symbols>;
  to: (keyOrChain: ChainKeys | Chain<ChainKeys>) => XcmSdkWithdrawTo<Symbols>;
```

Reply:
Oh awesome! I'll update the docs. Thanks!